src/EditorFeatures/Core/DocumentationComments/DocumentationCommentSuggestion.cs

```diff
@@ -65,26 +65,39 @@ public override Task OnProposalUpdatedAsync(SuggestionSessionBase session, Propo
             return Task.CompletedTask;
         }
 
-        public async Task TryDisplaySuggestionAsync(CancellationToken cancellationToken)
+        public async Task<bool> StartSuggestionSessionAsync(CancellationToken cancellationToken)
         {
-            _suggestionSession = await SuggestionManager.TryDisplaySuggestionAsync(this, cancellationToken).ConfigureAwait(false);
+            await RunWithEnqueueActionAsync(
+                "StartWork",
+                async () => _suggestionSession = await SuggestionManager.TryDisplaySuggestionAsync(this, cancellationToken).ConfigureAwait(false),
+            cancellationToken).ConfigureAwait(false);
 
-            if (_suggestionSession != null)
+            if (_suggestionSession is null)
```

Comment:
can you make RunWithEnqueueActionAsync return a value so you can keep the original `_suggestionSession = await RunWithEnqueueActionAsync(...)`?

Reply:
I don't think, with how the SuggestionManager.EnqueueAction doesn't return a value, that would make much sense.

Reply:
but we own RunWithEnqueueActionAsync